hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseTermReadSvcImpl.java

```
@@ -2489,7 +2552,26 @@ static boolean isDisplayLanguageMatch(String theReqLang, String theStoredLang) {
 		// NOTE: return the designation when one of then is not specified.
 		if (theReqLang == null || theStoredLang == null)
 			return true;
-		
+
 		return theReqLang.equalsIgnoreCase(theStoredLang);
     }
+
+
+	@Override
+	public Optional<IBaseResource> readCodeSystemByForcedId(String theForcedId) {
+		@SuppressWarnings("unchecked")
+		List<ResourceTable> resultList = (List<ResourceTable>) myEntityManager.createQuery(
```

Can you replace this with a Spring Data dao method?

---

I think if you make it return Optional it should throw an exception if 2 are returned (but I'm not 100% sure)

---

I still need the conversion performed by the toResource method. I could still move the query to a dao but didn't find a place I liked for it.
Also, yes I could get a data exception if more than one is returned by using getSingleResult instead of getResultList, but I thought it was a good chance to throw a more detailed exception for that case, as I saw somewhere else in the code that that use case being considered (constraints being removed), which is pretty scary.
Where would you move it?

---

ok makes sense.  just wanted to ensure you knew that was an option